### ISSUE TITLE
Make it easier to create demo previews with specific form-js branch

### ISSUE THREAD

**Issue by pinussilvestrus** [camunda/form-playground#16]
__What should we do?__

Creating previews with a specific form-js branch takes quite a set of manual work, currently.

![image](https://user-images.githubusercontent.com/9433996/201312613-bd6ad267-53bd-4992-8006-6570cfa3e475.png)

E.g. for [this fix](https://github.com/bpmn-io/form-js/pull/364) in the form-js-playground

* Create a branch in `camunda/form-playground` to [update the reference in the `link-deps` task](https://github.com/camunda/form-playground/blob/main/tasks/link-deps.js#L16)
* Add this branch to the list of deployable branches in [Netlify](https://app.netlify.com/) (currently via my personal account)
* Create a deployment in Netlify
* Share the URL manually: https://demo-submit-panel--camunda-form-playground.netlify.app/

We should
* [ ] move preview project from personal (@pinussilvestrus) to Netlify Pro team account  
* [ ] automate our preview environment pipelines (via Netlify)
* [ ] update [our handbook](https://github.com/camunda/team-hto) to cover the preview environment

There are a couple of things we can consider to improve the workflow

* for specific `form-js` PRs (`deploy-preview` label), automatically create a branch in `camunda/form-playground`
  * => This can be done via GitHub action
* auto-deploy specific branches `camunda/form-playground`
  * via Netlify
    * then we'd need a company account
    * **alternative**: do not auto-deploy, but use a [GitHub action](https://github.com/marketplace?type=actions&query=netlify+) to create a deployment
      * for this, we might not even need a branch, but simply replace the [`branch` config](https://github.com/camunda/form-playground/blob/main/tasks/link-deps.js#L16) via environment variable
      * Netlify offers [custom build hooks](https://docs.netlify.com/configure-builds/build-hooks/) for this
* post preview URL automatically back in `form-js` PR
  * => since the deployment is done via `camunda/form-playground`, how to connect back?
  * => a solution would be to trigger the deploy from `form-js` in `camunda/form-playground` and wait for the response
 
![image](https://user-images.githubusercontent.com/9433996/203948430-5f5152e5-29e7-4ee0-9bd2-d6ddb4d3e4bb.png)

Alternatives we had consider
* via [Cloudflare Pages](https://pages.cloudflare.com/) (cf. https://github.com/bpmn-io/form-js/pull/364#issuecomment-1284192425)
* via Infra Preview Environment (cf. https://github.com/camunda/form-playground/issues/16#issuecomment-1305538352)
  * => much overhead for a simple web app
* via GitHub pages (https://github.com/marketplace/actions/deploy-to-github-pages)

Inspiration: https://github.com/camunda/tasklist/blob/master/.github/workflows/preview-env-deploy.yml
Help desk issue: https://helpdesk.camunda.com/support/tickets/7382

__Why should we do it?__

Make branch previews for the playground easier available.

**Comment by pinussilvestrus** [comment]
I triggered a conversation with the Infra team: https://camunda.slack.com/archives/C5AHF1D8T/p1667299821412229

**Comment by pinussilvestrus** [comment]
Requirements we'd need to deliver once we decide to use a [Preview Environment](https://confluence.camunda.com/display/HAN/Preview+Environments) provided from the Infra team:
* Docker image of [the demo](https://github.com/camunda/form-playground/tree/main/example) that serves HTTP traffic via known port
* PR-based workflow
* orga overhead to establish secrets management, Docker image storage + permissions

**Comment by pinussilvestrus** [comment]
Another option (posted via [Slack](https://camunda.slack.com/archives/C5AHF1D8T/p1667912360987749?thread_ts=1667473129.607669&cid=C5AHF1D8T))

> did you consider Github Pages for that? Might be easier than wrapping frontend assets into a Docker image and then run that on K8s (lots of overhead) GH Pages also offers authentication if that is important to you

**Comment by pinussilvestrus** [comment]
How I would imagine it:

![image](https://user-images.githubusercontent.com/9433996/201323747-361e2c7d-4a9d-4e75-a2a0-704798fd88fd.png)


The first option creates a couple of individual branches in another repo (`camunda/form-playground`) that might not be needed and might never be cleaned up. For the second option, we'd need to find out how to trigger custom deployments with environment variables in our to-be-chosen provider.

**Comment by pinussilvestrus** [comment]
**Sync with @christian-konrad**
* We want to do the automation via GitHub action(s) and Netlify. It just works; no need to have extra effort to hook into the Infra-provided preview environment (e.g. via Docker container)
* We live with extra branches in `form-playground`. We anyway experienced in the past that we might need to make custom changes for an individual form-js feature in the Playground (for example datetime picker styling), so it's good to have the flexibility.
* To achieve that, we need to
  * [Request new IT Tool ](https://confluence.camunda.com/pages/viewpage.action?pageId=108431481)([Netlify Pro Account](https://www.netlify.com/pricing/), $19 per member/month)
  * Create the two GitHub actions to connect `form-js` to Netlify (cf. [this workflow](https://github.com/camunda/form-playground/issues/16#issuecomment-1311531824))

**Comment by pinussilvestrus** [comment]
I had also a small look into [Cloudflare pages](https://pages.cloudflare.com/), as we already have a company account over there (which would save us the [request process](https://confluence.camunda.com/pages/viewpage.action?pageId=108431481)) => cf. this [build configuration](https://dash.cloudflare.com/8c6170b66b8da3dd8917a72d6ab96687/pages/view/form-playground/9144a8fc-ea18-48ec-96b9-df31e8200ad2).

* In general, it looks quite similar with Netlify, with less configuration options
* There is a problem with `yarn`, as it seems it's not pre-installed

```text
* 16:19:38.678	Unable to link bpmn-io/form-js#develop due to error: Error: Command failed: /bin/sh -c yarn link
* 16:19:38.679	/bin/sh: 1: yarn: not found
```
* There is an [official GitHub action](https://github.com/marketplace/actions/cloudflare-pages-github-action) to trigger the deployment
  * However, it's not possible to specify the `FORM_JS_BRANCH` or a custom build command via this action, so it's not fitting our needs
  
TLDR: From what it looks like, we can't use this for our two-project use case, at least not without some overhead (e.g. creating our own Cloudflare Pages GitHub action).

**Comment by pinussilvestrus** [comment]
Update on Purchasing Netlify Pro, please find the [Helpdesk Issue here.](https://helpdesk.camunda.com/support/tickets/7382): We are currently waiting for the Netlify Privacy team to give some necessary information on their policies.

**Comment by pinussilvestrus** [comment]
Here is a little update on this one:
* We are currently waiting for Netlify to send the NDA to our legal department to sign it
* Winter break slowed down things a lot
* In general: I don't feel we are slow on our side, but on Netlify's side, it seems it takes loooong to provide the necessary information + unclear responsibilities on their side. Every time it seems I talk to another person who is in charge. Let's hope they finally manage to send us the NDA this week so we can proceed.

**Comment by pinussilvestrus** [comment]
The security docs were approved from our side 👍 

Cf. https://helpdesk.camunda.com/support/tickets/7382

```text
InfoSec Review: approved
They provided their latest SOC2 T2 report which has no findings.
Likelihood: unlikely (2)
CI Confidentiality: Confidential (8)
```

Waiting for next steps to continue the purchase.

**Comment by pinussilvestrus** [comment]
The purchase was now formally requested: https://helpdesk.camunda.com/support/tickets/8232.

We are almost there 🚀

**Comment by pinussilvestrus** [comment]
We now have a team account on Netlify; I shared the credentials on our vault. 

I plan to reserve some time this week to proceed with creating the pipelines.

**Comment by pinussilvestrus** [comment]
I just moved [the demo](https://camunda-form-playground.netlify.app/) from my personal Netlify account to our new HTO team ✅
